climada/entity/impact_funcs/test/test_base.py

```diff
@@ -60,6 +59,50 @@ def test_from_sigmoid(self):
         self.assertEqual(imp_fun.haz_type, 'RF')
         self.assertEqual(imp_fun.id, 2)
 
+    def test_from_poly_s_shape(self):
+        """Check default impact function: polynomial s-shape"""
+
+        haz_type = 'RF'
+        threshold = 0.2
+        half_point = 1
+        scale = 0.8
+        exponent = 4
+        impf_id = 2
+        unit = 'm'
+        intensity = (0, 5, 5)
+
+        def test_aux_vars(impf):
+            self.assertTrue(np.array_equal(impf.paa, np.ones(5)))
+            self.assertTrue(np.array_equal(impf.intensity, np.linspace(0, 5, 5)))
+            self.assertEqual(impf.haz_type, haz_type)
+            self.assertEqual(impf.id, impf_id)
+            self.assertEqual(impf.intensity_unit, unit)
+
+        impf = ImpactFunc.from_poly_s_shape(
+            intensity=intensity, threshold=threshold, half_point=half_point, scale=scale,
+            exponent=exponent, haz_type=haz_type, impf_id=impf_id, intensity_unit=unit
+            )
+        # True value can easily be computed with a calculator
+        correct_mdd = np.array([0, 0.59836395, 0.78845941, 0.79794213, 0.79938319])
+        np.testing.assert_array_almost_equal(impf.mdd, correct_mdd)
+        test_aux_vars(impf)
+
+        # If threshold > half_point, mdd should all be 0
+        impf = ImpactFunc.from_poly_s_shape(
+            intensity=intensity, threshold=half_point*2, half_point=half_point, scale=scale,
+            exponent=exponent, haz_type=haz_type, impf_id=impf_id, intensity_unit=unit
+            )
+        np.testing.assert_array_almost_equal(impf.mdd, np.zeros(5))
+        test_aux_vars(impf)
+
+        with self.assertRaises(ValueError):
```

Comment:
Check actual error
```suggestion
        with self.assertRaisesRegex(ValueError, "Exponent value"):
```

Reply:
Thanks for the correction! Why should we priorize using `assertRaisesRegex` ?

Reply:
Because it tests the actual error message. `assertRaises` would succeed with *any* `ValueError` thrown, but you want to test for a specific one here.

Note that this is basically a shortcut for the idiom we were using so far:
```python
with self.assertRaises(Exception) as cm:
    do_stuff()
self.assertIn("Text", str(cm.exception))
```

Reply:
Makes total sense, thanks!